### ISSUE TITLE
Atom 1.2.0 (1.3.0-beta) rendering is slow

### ISSUE THREAD

**Issue by alexbeletsky** [atom/atom#9561]
Sorry, for such subjective and non-technical bug description.

I've installed Atom 1.2.0 (as well as Atom 1.3.0-beta) yesterday. Release notes saying about performance improvements (both rending and startup time) that are hardly could be noticed.

1.2.0 (and 1.3.0-beta) rendering is very slow. What I mean, if I click some file, I first see the "grey" text and only after the coloring syntax applies. So, it blinks and for me it's really hurting experience.

Had to downgrade to 1.1.0 again. If I can provide more information, please let me know.

**Comment by benogle** [comment]
To be clear, it is only slower when you are initially loading a file? Or is it slower in other areas as well? What kind of files are you loading? How large? On a fresh window open? Or is it slower the longer the window is open?

Can you try starting atom in [safe mode](https://atom.io/docs/latest/hacking-atom-debugging#check-if-the-problem-shows-up-in-safe-mode)?

If safe mode is faster, can you paste in the contents of `apm list`?

**Comment by alexbeletsky** [comment]
@benogle 

1. Yes, it's slow on initial file rendering. With 1.1 I'm having the same effect, so then I open the file it always "blinks", but on 1.2 the blink is much longer, so it's really annoying.

2. My files are typically `.js` (Javascript-JSX), `.scss` and `.html` relatively low size.

3. I've tried safe mode. I still see a little blinks, but in general it works much more faster. 

My [apm list](https://gist.github.com/alexbeletsky/20e2da5f02b97b564e50).

**Comment by patoi** [comment]
I have same issue with 1.2 on UHD (3840x2160), but not on HD (1920x1080). One laptop with dual screen HD + UHD (HDMI1.2@30Hz).

**Comment by patoi** [comment]
There are some CPU and TimeLine data from DevTools (zip 2.8MB):
https://drive.google.com/open?id=0B6pleSdQXj0YV245UXc4ZVQxblE

**Comment by benogle** [comment]
@alexbeletsky ok, it may be a package slowing things down if it's faster in safe mode. 

My guesses for culprits would be: minimap, pigments, linter, with enhanced-tabs, and editorconfig coming in second. Can you try disabling them one by one and reloading? Is there a difference?

**Comment by patoi** [comment]
I've turned off all Community Packages, and all Core packages but it is slow rendering on UHD.
(settings-view, tabs and tree-view enabled)

video: https://drive.google.com/open?id=0B6pleSdQXj0Ya1FCTVZHbENHT2M

config.cson:

  disabledPackages: [
      "atom-jasmine"
      "atom-jade"
      "gitignore-snippets"
      "language-groovy"
      "language-ruby"
      "language-ruby-on-rails"
      "language-php"
      "language-perl"
      "language-objective-c"
      "markdown-preview"
      "minimap"
      "javascript-snippets"
      "language-asciidoc"
      "linter"
      "linter-jshint"
      "lodash-snippets"
      "merge-conflicts"
      "angularjs"
      "atom-beautify"
      "atom-bootstrap3"
      "asciidoc-preview"
      "atom-jshint"
      "docblockr"
      "editorconfig"
      "new-fast-duplicate"
      "project-manager"
      "script"
      "xml-formatter"
      "language-go"
      "language-java"
      "language-make"
      "language-csharp"
      "language-coffee-script"
      "language-clojure"
      "language-c"
      "about"
      "archive-view"
      "autocomplete-atom-api"
      "autocomplete-css"
      "autocomplete-html"
      "autocomplete-plus"
      "autocomplete-snippets"
      "autoflow"
      "autosave"
      "background-tips"
      "bookmarks"
      "bracket-matcher"
      "command-palette"
      "deprecation-cop"
      "dev-live-reload"
      "encoding-selector"
      "find-and-replace"
      "language-css"
      "language-gfm"
      "language-git"
      "language-html"
      "language-hyperlink"
      "language-javascript"
      "language-json"
      "language-less"
      "language-mustache"
      "language-property-list"
      "language-python"
      "language-sass"
      "language-shellscript"
      "language-source"
      "language-sql"
      "language-text"
      "language-todo"
      "language-toml"
      "language-xml"
      "language-yaml"
      "line-ending-selector"
      "link"
      "notifications"
      "open-on-github"
      "package-generator"
      "release-notes"
      "fuzzy-finder"
      "go-to-line"
      "git-diff"
      "grammar-selector"
      "image-view"
      "incompatible-packages"
      "keybinding-resolver"
      "snippets"
      "spell-check"
      "status-bar"
      "styleguide"
      "symbols-view"
      "wrap-guide"
      "whitespace"
      "welcome"
      "update-package-dependencies"
      "timecop"
    ]

**Comment by patoi** [comment]
Ubuntu box.

cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=15.10
DISTRIB_CODENAME=wily
DISTRIB_DESCRIPTION="Ubuntu 15.10"

uname -a
Linux patoi-y510p 4.2.0-18-generic #22-Ubuntu SMP Fri Nov 6 18:25:50 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

**Comment by gerardroche** [comment]
Are you talking about this:

![504f5626-f801-11e3-85a4-6efb438676bd](https://cloud.githubusercontent.com/assets/44148/11483915/130ffa8e-97a3-11e5-9294-fd678651f2f7.gif)

re: https://github.com/atom/atom/issues/2679

If yes, this isn't a issue just in recent versions. It has been an issue since always. It's a [FOUC](https://en.wikipedia.org/wiki/Flash_of_unstyled_content), a Flash of Unstyled Content.

**Comment by patoi** [comment]
This is not the same issue gerardroche

I've uploaded the video: https://youtu.be/kpf362RJkM4

It is working fine on HD screen, but on UHD screen (see details above) Atom is "painting" the screen.

New information:
* It is the problem with 1.2.0 - 1.2.4
* When I click on a tab, and back, then the "painting" has occured too.

**Comment by alexbeletsky** [comment]
@gerardroche yes, that's exactly the experience I have.. I don't know, but in 1.1 that was not that visible for me, but in 1.2 FOUC hit me hard. To be honest, that was totally unacceptable for me (with each file opened I was distracted by that effect) and switched back to ST.

the problem that @patoi is showing looks like a showstopper on UHD monitors. What's your setup exactly (type of video card, type of monitor)? UHD monitors are not so well spreaded at the moment, but it's just a questions of time (I'm thinking to get one, also).

**Comment by patoi** [comment]
Monitor: LG 40UB800VA-ZA
Laptop: Lenovo y510p
GPU: 01:00.0 VGA compatible controller: NVIDIA Corporation GK107M [GeForce GT 755M] (rev ff)
Driver: NVidia binary driver 352.63 (proprietary, tested)

Monitor is running on 30Hz, because the laptop hdmi output is 1.4

**Comment by fiws** [comment]
UHD monitor with no issues here. I think the FOUC is worse than in 1.1, but still acceptable and not like in @patoi s video. 


PS: @patoi might want to try [Simple Screen Recorder](http://www.maartenbaert.be/simplescreenrecorder/) for the video next time, its super easy ;)

**Comment by gerardroche** [comment]
For me the FOUC isn't anything new. And it is totally unacceptable. It's one of the few blockers for me. It probably makes performance seem worse than actually might be.

**Comment by patoi** [comment]
There is a video capture: https://youtu.be/e5-hFEIOAFM

( @fiws thanks for the tip )

**Comment by patoi** [comment]
There is a 'major GC' 2.044s when I changing the tab.

![atom-uhd-img-01](https://cloud.githubusercontent.com/assets/1846548/11539231/64ff4190-9926-11e5-8486-590de025a960.png)

**Comment by thedaniel** [comment]
That's only 2 milliseconds, so probably not related.

**Comment by patoi** [comment]
Oh, this is my mistake.

**Comment by patoi** [comment]
Today I've realized that if I don't maximize the screen (but it is near the maximum size) it is working fine. If I maximize the screen then the slowing turns up.

**Comment by arturoc** [comment]
same here in ubuntu 15.10 using the unity desktop manager. 

if the window is maximized you can see the vertical refresh of the window and in general everything related to updating the ui is really sluggish, if it's just as big but not maximized the refresh is instantaneous. this didn't happen in 1.0. i'm using an intel graphics card in case that gives any clue

**Comment by drewzh** [comment]
I'm also experiencing extremely bad performance with my brand new XPS 9550 running on Intel i915. I experience the same issues as documented in the videos above. When I switch to using the nvidia GPU with proprietary drivers the performance is fine again, but I have sections of menus etc that don't render properly. Basically I just can't get atom to function nicely on this laptop. I've tried Fedora + Ubuntu latest releases so far, both distros (running Gnome Shell) show the same symptoms.

On a side note, Chrome is showing rendering issues also. I've had to install the latest unstable chrome and that seems to have fixed Chrome at least.

**Comment by lhavel** [comment]
exactly the same issue here on a Dell XPS 15 9530 laptop (UHD) and Ubuntu Gnome 15.10 with Nouveau driver.
I get the very same behavior as described by @patoi and also experiencing Chrome rendering issues as described by @drewzh.

**Comment by fuxialexander** [comment]
I can confirm similar issue also appears on OS X. My setup is 2015 Macbook pro 13' (no Nvidia card) with DELL P2415Q (4K). The scrolling speed with a small window size is much more faster (or normal) than that in fullscreen. Sublime text seems do not have that issue. The VS Code is slow down a little bit, but feels better than atom.

**Comment by YemSalat** [comment]
Chromium has been slow for me on Linux Mint  since about 3 versions ago. It seems like Atom may have 'inherited' that slowness.

**Comment by EZWrighter** [comment]
I can confirm this issue with Ubuntu 15.10 with Atom 1.2.4 using 4k monitor.  If the window is fullscreen, you get horrible repaint times.  If the window is not fullscreen (Any Size), the repaint times are as expected.

**Comment by patoi** [comment]
Atom 1.4.0 has affected.

**Comment by YemSalat** [comment]
Same here, I think it would be helpful if everyone who's replying to this thread would mention which operating system they are using, seems like its mostly Debian-based linux distros are affected (Ubuntu, Mint, etc.).
PS Do you also notice the same lag when running Chromium?

**Comment by EZWrighter** [comment]
This seems related if not a duplicate of https://github.com/atom/atom/issues/9724

**Comment by patoi** [comment]
I've installed new graphic driver from ppa, as described in #9724
Atom is working with faster rendering on full UHD screen, but I've got some font/line rendering issue with the new ppa driver on the all window sizes.

When I'm moving cursor up or down at the first column, the font rendering is weird.

Screenshots, line rendered weird at red dots. Last image contains the default, good rendered line.

![atom_ubuntu_bad_rendering_01](https://cloud.githubusercontent.com/assets/1846548/12671393/a7823722-c66f-11e5-9ed3-3140c924f82d.png)
![atom_ubuntu_bad_rendering_02](https://cloud.githubusercontent.com/assets/1846548/12671391/a77f5e4e-c66f-11e5-9b1a-65ca2d563daa.png)
![atom_ubuntu_good_rendering](https://cloud.githubusercontent.com/assets/1846548/12671392/a781afd2-c66f-11e5-988e-db6e6e39289a.png)

**Comment by leifcr** [comment]
@patoi I'm seeing the same behaviour here, so a change in the driver between December 11th 2015 and mid January.

I'm suspecting this commit, since it looks like a sync issue: http://cgit.freedesktop.org/xorg/driver/xf86-video-intel/commit/?id=963b5bb1be874623f06caa421e06cdd2134204dc

However I haven't been able to test it.

**Comment by thegeologician** [comment]
I can confirm the same behaviour @patoi posted, similar to his video, just not quite as slow. UI rendering (opening files, preferences, integrated terminal...) is really  sluggish in fullscreen, it's fine at any non-maximised window size.
Dell M3800, 3840x2160, using the integrated Intel Graphics, not the Nvidia card
Fedora 23 Workstation
Atom 1.4.3
Chrome does not experience (and never has) perceivable rendering issues.

**Comment by thegeologician** [comment]
Additional to my previous post, here a screenshot of the profiler, as recommended in the atom docs:
![screenshot from 2016-02-06 22-58-55](https://cloud.githubusercontent.com/assets/944349/12869945/03c4dc26-cd26-11e5-94ac-13bc6954932b.png)

Also, I'm seeing these messages on the console, not sure if that's related:
![screenshot from 2016-02-06 23-00-54](https://cloud.githubusercontent.com/assets/944349/12869952/1df44000-cd26-11e5-920a-3b0133bba8c5.png)

Behaviour is the same in safe mode.

**Comment by leifcr** [comment]
@patoi Regarding the flickering: see #10700

**Comment by icelight0** [comment]
As i wrote in #10518 i'm having the same issue on arch with i915 and an uhd screen. I don't have any issues when running chrome, but vivaldi, which is also based on chromium embedded, has the same draw issues.

**Comment by Oblik-Design** [comment]
same issues here, super slow, on a 32 inch 4k display, next to my 2015 macbook pro with 2 graphics cards. shuts down after a minute. 
![capture d ecran 2016-02-09 a 21 29 53](https://cloud.githubusercontent.com/assets/9968597/12937260/ea43af7e-cf74-11e5-9024-8bd7c3843331.png)

**Comment by arturoc** [comment]
just to point out that this doesn't happen anymore on ubuntu 16.04

**Comment by kumarharsh** [comment]
I have a project with more than 1100 files, and atom can't even open the project anymore. It's actually tree-view which locks up it seems. If some days atom *does* manage to open up the project (after locking up 2-3 times), when I go to the tree view and click on the arrow button to expand a folder, atom crashes. I can't use atom anymore like this.

**Comment by fiws** [comment]
@kumarharsh probably another issue, maybe create a separate one?

**Comment by Oblik-Design** [comment]
It has been better, no more slow down, but I must admit passing to brackets while Atom was not rendering well on my 4k monitors.... I wish I could come back, but bracket-snippets plugin is just incredible, I cannot live without it , no more json files to create... I ll be back to Atom the day when you'll have a better snippet manager.

**Comment by axe-z** [comment]
it's been fixed.
